ambari-server/src/main/java/org/apache/ambari/server/state/StackInfo.java

```diff
@@ -513,27 +513,44 @@ public Map<String, PropertyInfo> getRequiredProperties() {
   }
 
   public Map<PropertyInfo.PropertyType, Set<String>> getConfigPropertiesTypes(String configType) {
-    if(!propertiesTypesCache.containsKey(configType)) {
+    if (!propertiesTypesCache.containsKey(configType)) {
       Map<PropertyInfo.PropertyType, Set<String>> propertiesTypes = new HashMap<>();
+
+      collectPropertyTypes(propertiesTypes, getProperties(), configType);
```

Comment:
This line is need to properly add property types for configurations at stack root (e.g. cluster-env.xml).